test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentSkipListSetTest.scala

```diff
@@ -0,0 +1,423 @@
+      expect(csls.isEmpty).toBeTruthy
+      expect(csls.remove(333)).toBeFalsy
+      expect{
+          try {
+            csls.first
+            false
+          } catch {
+            case _ : NoSuchElementException => true
+            case _ : Throwable => false
+          }
+      }.toBeTruthy
+    }
+
+    it("should store and remove ordered strings") {
+      val csls = new ConcurrentSkipListSet[String]()
+      
+      expect(csls.size()).toEqual(0)
+      expect(csls.add("222")).toBeTruthy
+      expect(csls.size()).toEqual(1)
+      expect(csls.add("111")).toBeTruthy
+      expect(csls.size()).toEqual(2)
+      expect(csls.first).toEqual("111")
+      expect(csls.remove("111")).toBeTruthy
+      
+      expect(csls.size()).toEqual(1)
+      expect(csls.first).toEqual("222")
+
+      expect(csls.remove("222")).toBeTruthy
+      expect(csls.size()).toEqual(0)
+      expect(csls.remove("333")).toBeFalsy
+      expect(csls.isEmpty).toBeTruthy
+    }
+
+    when("compliant-asinstanceofs").
+    it("should throw exception on non comparable objects") {
+      case class TestObj(num: Int) 
+
+      val csls = new ConcurrentSkipListSet[TestObj]()
+      
+      expect(csls.size()).toEqual(0)
+      expect(csls.add(TestObj(222))).toBeTruthy
+      expect(csls.size()).toEqual(1)
+      expect{
+        try {
+          csls.add(TestObj(111))
+          false
+        } catch {
+          case _ : ClassCastException => true
+          case _ : Throwable => false
+        }
+      }.toBeTruthy
+      expect(csls.comparator eq null).toBeTruthy
+    }
+    
+    it("should store objects with custom comparables") {
+      case class Rect(x: Int, y: Int)
+
+      val areaComp = new Comparator[Rect] {
+        def compare(a: Rect, b: Rect): Int = (a.x*a.y) - (b.x*b.y)
+      }
+
+      val csls = new ConcurrentSkipListSet[Rect](areaComp)
+      
+      expect(csls.add(Rect(1,2))).toBeTruthy
+      expect(csls.add(Rect(2,3))).toBeTruthy
+      expect(csls.add(Rect(1,3))).toBeTruthy
+      
+      val first = csls.first()
+      expect(first.x).toEqual(1)
+      expect(first.y).toEqual(2)
+
+      expect(csls.remove(first)).toBeTruthy
+      expect(csls.remove(first)).toBeFalsy
+
+      val second = csls.first()
+      expect(second.x).toEqual(1)
+      expect(second.y).toEqual(3)
+
+      expect(csls.remove(second)).toBeTruthy
+
+      val third = csls.first()
+      expect(third.x).toEqual(2)
+      expect(third.y).toEqual(3)
+
+      expect(csls.remove(third)).toBeTruthy
+
+      expect(csls.isEmpty).toBeTruthy
+    }
+
+    it("should store ordered Double even in corner cases") {
+      val csls = new ConcurrentSkipListSet[Double]()
+      
+      
+      expect(csls.add(1.0)).toBeTruthy
+      expect(csls.add(+0.0)).toBeTruthy
+      expect(csls.add(-0.0)).toBeTruthy
+      expect(csls.add(Double.NaN)).toBeTruthy
+      
+      expect(csls.first).toEqual(-0.0)
+      expect(csls.first.hashCode).toEqual(-0.0.hashCode)
```

Comment:
Ah ah ^^

It's a bit obscure, as a test, though. Consider `expect(csls.first.equals(-0.0)).toBeTruthy` instead.